investigations/FastestResponseTime/app/src/main/java/org/mozilla/perf/investigations/MainActivity.kt

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.perf.investigations
+
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.os.SystemClock
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.PopupWindow
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.*
+import org.mozilla.perf.investigations.DurationTimes.duration
+
+private val uiHandler = Handler(Looper.getMainLooper())
+
+/**
+ * See <proj-root>/README.md for details on this experiment.
+ */
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var preinflatedPopupWindowLayout: View
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        preinflatedPopupWindowLayout = LayoutInflater.from(this).inflate(R.layout.popup_window_preinflated, null, false)
+
+        popupWindowButton.setOnClickListener {
+            Log.e("lol", "popupWindowButton.onClick")
+            DurationTimes.start = SystemClock.elapsedRealtime()
+            showPopupWindow()
+        }
+
+        popupWindowPreinflatedButton.setOnClickListener {
+            Log.e("lol", "popupWindowPreinflatedButton.onClick")
+            DurationTimes.start = SystemClock.elapsedRealtime()
+            showPopupWindowPreinflated()
+        }
+    }
+
+    private fun showPopupWindow() {
+        val layout = LayoutInflater.from(this).inflate(R.layout.popup_window, null, false)
+
+        val window = PopupWindow(layout, 500, 500, true)
+        window.showAsDropDown(popupWindowButton)
+        window.showAtLocation(contentContainer, 0, 200, 200)
+
+        // For an unknown reason, the onDraw method never gets called if this is attached to the layout
+        // in the PopupWindow so we set it on the contentContainer instead.
+        contentContainer.viewTreeObserver.addOnDrawListener(DrawOnce(contentContainer, "PopupWindow duration"))
+
+        // Apparently tapping outside a popup window to dismiss is custom behavior so we do the
+        // simple thing and hide them after a few seconds
+        uiHandler.postDelayed({
+            window.dismiss()
```

Comment:
I don't quite get the action of automatically dismissing it because clicking outside is custom behavior. Do we want to simulate closing of the popup too?

Reply:
The pop-up covers the content of the app so, in order to re-use the app without restarting it, we have to dismiss the pop-up or restart the app. afaict, this is the simplest way to dismiss the pop-up.

I can add to the comment to clarify.

Reply:
Ah that makes sense. I have nothing against it ,it wasn't clear to me why we needed it to dismiss. Sounds good to me, we're not testing dismissing properties anyway.